components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/client/model/AuthenticationRequestWrapper.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.com).
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authentication.endpoint.util.client.model;
+
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+
+/**
+ * Authentication endpoint request wrapper.
+ */
+public class AuthenticationRequestWrapper extends HttpServletRequestWrapper {
+    private Map<String, Object> cachedParams;
```

Comment:
cache has a specific meaning?

Are these actually cached parameters?

Reply:
Used `cachedParams` since we filter these parameters from the URL and retrieve using API. Alternatively we can use `authParams`.